main/asterisk.c

```diff
@@ -3172,26 +3172,46 @@ static int ast_el_read_history(const char *filename)
 	return history(el_hist, &ev, H_LOAD, filename);
 }
 
+static char *get_default_histfile_name(void)
```

Comment:
Just a thought - have this take a function pointer (`int (*readwrite)(const char *filename)`) and pass in either `ast_el_read_history` or `ast_el_write_history` depending on where you call it from. Then you can put all the memory allocation & freeing in one place rather than relying on the caller to do it. Something like:

```
/* rename "get_default_histfile_name" to "process_histfile" or something like that */

static void ast_el_read_default_histfile(void)
{
    process_histfile(ast_el_read_history);
}

static void ast_el_write_default_histfile(void)
{
    process_histfile(ast_el_write_history);
}
```